core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -871,7 +871,7 @@ private[deploy] class Master(
         logInfo("Telling app of decommission executors")
         exec.application.driver.send(ExecutorUpdated(
           exec.id, ExecutorState.DECOMMISSIONED,
-          Some("worker decommissioned"), None, workerLost = false))
+          Some("worker decommissioned"), None, workerLost = true))
```

Comment:
Why did this need to change to true?

Reply:
The short answer is that I sort of hacked it here :-)

So on line 186 of StandaloneAppClient.scala in this PR: I actually plumb the workerLost (= true) here into ExecutorDecommissionInfo.isHostDecommissioned. 

The proper thing to do here would be to not communicate executor decommissioning via ExecutorUpdated in the standalone codepath. But instead have a separate ExecutorDecommissioned message that can take the ExecutorDecommissionInfo. The ExecutorDecommissionInfo should be prepared by the cluster manager directly, since it has the most context about the decom is happening (ie exactly what pieces would be going away). 

This is just a slightly bigger change and maybe I should just do it here. Let me try it.

Reply:
On second thoughts, this is a bigger change: I will just explain why workerLost = true better in the code. 

The main problem is that the Deploy messages are b/w the Master and the Driver, while the ExecutorDecommissionInfo are b/w the Driver and the Executors. In any case, each cluster manager would need to have its own way of somehow communicating the decommissioning metadata to the driver, and this workerLost could be the Master's internal way.

The alternatives would be to either introduce another `ExecutorDecommissioned(id, MasterToDriverExecutorDecommissionInfo)` struct, or enhance `ExecutorUpdated.workerLost` to be a full Struct/Any that can take the MasterToDriverExecutorDecommissionInfo. 

This all seems like a bit too much machinery for now, so I will just comment it inline in the code. 

Thank you for raising this question :-)